content/modules/dev/ui.docView.collage.js

```diff
@@ -1,502 +1,539 @@
+            var st = $elt.scrollTop();
+            var H = $elt.parent().height();
+
+            //pbar is inititialized to "first guess" of current page, assuming all heights are equal...
+            var pbar = Math.max(1, Math.min(Math.ceil(self._collection.items.length * st / H), self._collection.items.length));
+            var p = $('div.material[page=' + pbar + ']', self.element);
+            var next_p;
+            var cond = p.offset().top < 0;
+            while (p.offset().top < 0 === cond) {
+              next_p =  p[cond ? 'next' : 'prev']();
+              if (next_p[0].hasAttribute('page')) {
+                p = next_p;
+              }              else {
+                break;
+              }
             }
+
+            if (p.offset().top < 0) { //make sure we get page with smallest but positive "top" value
+              p = p.next();
             }
+
+            $.concierge.trigger({ type: 'page_peek', value:p.attr('page') });
+          }, 300);
+
+          self._scrollTimerID =  timerID;
         }
-        else if (action === "remove" && items_fieldname === "location"){ //just re-render the pages where locations were just removed. 
-            D        = payload.diff;
-            var pages_done    = {};
-            
-            for (i in D){
-            page = D[i].page;
-            if (! (page in pages_done)){
-                pages_done[page] = null;
-                delete this._pages[page];
-                this._render_one(page);
-            }
-            }
-        }        
-        }, 
-        _update: function(){
-        $.ui.view.prototype._update.call(this);
-        var self = this;
-        self._generate_contents();
-        self._scroll_to_page();
-        },       
-        close: function(){
-        var id =  this._id_collection;
-        delete $.concierge.features["doc_viewer"][id];
-        $.ui.view.prototype.close.call(this);
-        $.L("closing docviewer",  id);
-        },
-        _scroll_to_page: function(cb){
-        var self = this;
-        var st = self.element[0].scrollTop;
-        var page1 =  $("div.material[page="+self._page+"]", self.element);
-        var w1 = page1.width();
-        var w2 = self.element.width();
-        //        var pbar = Math.ceil((st+0.01)/(self._h+self._v_margin));
-        //        var current_page = self._page;
-        if (self._in_page_transition === false){
-            self._in_page_transition =  true;
-            var divOffset = self.element.offset().top;
-
-            var materialPage = $("div.material[page="+self._page+"]", self.element);
-            if (materialPage.offset() !== undefined) {
-               var imgOffset = materialPage.offset().top;
-               self.element.animate({scrollTop: '+=' + (imgOffset-divOffset) + 'px'}, 500, function(){
-                   self._in_page_transition =  false;
-               });
-            }
+      });
+
+      //REMOVED        self._update_best_fit_zoom();
```

Comment:
not sure if we should take this out or not since i have no idea what it does right now.